src/pages/home/sidebar/SidebarLinksData.tsx

```diff
@@ -1,149 +1,133 @@
 import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import SidebarLinks, {basePropTypes} from './SidebarLinks';
-
-const propTypes = {
-    ...basePropTypes,
-
-    /* Onyx Props */
+import type * as OnyxTypes from '@src/types/onyx';
+import type {Message} from '@src/types/onyx/ReportAction';
+import SidebarLinks from './SidebarLinks';
+
+type PickedReport = Pick<
+    OnyxTypes.Report,
+    | 'reportID'
+    | 'participantAccountIDs'
+    | 'hasDraft'
+    | 'isPinned'
+    | 'isHidden'
+    | 'notificationPreference'
+    | 'errorFields'
+    | 'lastMessageText'
+    | 'lastVisibleActionCreated'
+    | 'iouReportID'
+    | 'total'
+    | 'nonReimbursableTotal'
+    | 'hasOutstandingChildRequest'
+    | 'isWaitingOnBankAccount'
+    | 'statusNum'
+    | 'stateNum'
+    | 'chatType'
+    | 'type'
+    | 'policyID'
+    | 'visibility'
+    | 'lastReadTime'
+    | 'reportName'
+    | 'policyName'
+    | 'oldPolicyName'
+    | 'ownerAccountID'
+    | 'currency'
+    | 'managerID'
+    | 'parentReportActionID'
+    | 'parentReportID'
+    | 'isDeletedParentAction'
+>;
+
+type SidebarLinksDataOnyxProps = {
     /** List of reports */
-    chatReports: PropTypes.objectOf(reportPropTypes),
-
-    /** All report actions for all reports */
-
-    /** Object of report actions for this report */
-    allReportActions: PropTypes.objectOf(
-        PropTypes.arrayOf(
-            PropTypes.shape({
-                error: PropTypes.string,
-                message: PropTypes.arrayOf(
-                    PropTypes.shape({
-                        moderationDecision: PropTypes.shape({
-                            decision: PropTypes.string,
-                        }),
-                    }),
-                ),
-            }),
-        ),
-    ),
+    chatReports: OnyxCollection<PickedReport & {isUnreadWithMention: boolean}>;
 
-    /** Whether the reports are loading. When false it means they are ready to be used. */
-    isLoadingApp: PropTypes.bool,
+    /** Wheather the reports are loading. When false it means they are ready to be used. */
+    isLoadingApp: OnyxEntry<boolean>;
 
     /** The chat priority mode */
-    priorityMode: PropTypes.string,
+    priorityMode: OnyxEntry<ValueOf<typeof CONST.PRIORITY_MODE>>;
 
     /** Beta features list */
-    betas: PropTypes.arrayOf(PropTypes.string),
+    betas: OnyxEntry<OnyxTypes.Beta[]>;
 
-    network: networkPropTypes.isRequired,
+    /** All report actions for all reports */
+    allReportActions: OnyxEntry<Array<Pick<OnyxTypes.ReportAction, 'reportActionID' | 'actionName' | 'errors' | 'message'>>>;
 
     /** The policies which the user has access to */
-    // eslint-disable-next-line react/forbid-prop-types
-    policies: PropTypes.object,
-
-    // eslint-disable-next-line react/forbid-prop-types
-    policyMembers: PropTypes.object,
+    policies: OnyxCollection<Pick<OnyxTypes.Policy, 'type' | 'name' | 'avatar'>>;
```

Comment:
Same, I think it makes sense to reuse `policySelector` return type (create it if it's not there)